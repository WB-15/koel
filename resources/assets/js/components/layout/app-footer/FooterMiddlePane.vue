<template>
  <div class="middle-pane" data-testid="footer-middle-pane">
    <div id="progressPane" class="progress">
      <template v-if="song">
        <h3 class="title">{{ song.title }}</h3>
        <p class="meta">
          <a :href="`/#!/artist/${song.artist.id}`" class="artist">{{ song.artist.name }}</a> â€“
          <a :href="`/#!/album/${song.album.id}`" class="album">{{ song.album.name }}</a>
        </p>
      </template>

      <div class="plyr">
        <audio controls crossorigin="anonymous"></audio>
      </div>
    </div>
  </div>
</template>

<script lang="ts" setup>
import { toRefs } from 'vue'

const props = defineProps<{ song?: Song }>()
const { song } = toRefs(props)
</script>

<style lang="scss">/* no scoping here because we're overriding some plyr classes */
.middle-pane {
  flex: 1;
  display: flex;

  @media only screen and (max-width: 768px) {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    height: 8px;
  }
}

#progressPane {
  flex: 1;
  position: relative;
  display: flex;
  flex-direction: column;
  place-content: center;
  place-items: center;

  .meta {
    font-size: .9rem;

    a {
      &:hover {
        color: var(--color-highlight);
      }
    }
  }

  // Some little tweaks here and there
  .plyr {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
  }

  .plyr__progress {
    &--seek {
      height: 11px; // increase click area
    }
  }

  .plyr__controls {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    padding: 0;

    &--left, &--right {
      display: none;
    }
  }

  @media only screen and (max-width: 768px) {
    .meta, .title {
      display: none;
    }

    .plyr__progress {
      &--seek {
        border-bottom-color: var(--color-bg-primary);
      }
    }
  }
}
</style>
