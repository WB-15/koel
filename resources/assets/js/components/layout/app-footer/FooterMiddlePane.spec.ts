import { beforeEach, expect, it } from 'vitest'
import { render } from '@/__tests__/__helpers__'
import FooterMiddlePane from './FooterMiddlePane.vue'
import factory from '@/__tests__/factory'
import { cleanup } from '@testing-library/vue'

beforeEach(() => cleanup())

it('renders without a song', () => {
  expect(render(FooterMiddlePane).html()).toMatchSnapshot()
})

it('renders with a song', () => {
  const album = factory<Album>('album', {
    id: 42,
    name: 'IV',
    artist: factory<Artist>('artist', {
      id: 104,
      name: 'Led Zeppelin'
    })
  })

  const song = factory<Song>('song', {
    album,
    title: 'Fahrst√ºhl to Heaven',
    artist: album.artist
  })

  expect(render(FooterMiddlePane, {
    props: {
      song
    }
  }).html()).toMatchSnapshot()
})
