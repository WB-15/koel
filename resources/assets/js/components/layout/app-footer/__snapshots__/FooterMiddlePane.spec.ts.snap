// Vitest Snapshot v1

exports[`renders with a song 1`] = `
"<div class=\\"middle-pane\\" data-testid=\\"footer-middle-pane\\">
  <div id=\\"progressPane\\" class=\\"progress\\">
    <h3 class=\\"title\\">Fahrstühl to Heaven</h3>
    <p class=\\"meta\\"><a href=\\"/#!/artist/104\\" class=\\"artist\\">Led Zeppelin</a> – <a href=\\"/#!/album/42\\" class=\\"album\\">IV</a></p>
    <div class=\\"plyr\\"><audio controls=\\"\\" crossorigin=\\"anonymous\\"></audio></div>
  </div>
</div>"
`;

exports[`renders without a song 1`] = `
"<div class=\\"middle-pane\\" data-testid=\\"footer-middle-pane\\">
  <div id=\\"progressPane\\" class=\\"progress\\">
    <!--v-if-->
    <div class=\\"plyr\\"><audio controls=\\"\\" crossorigin=\\"anonymous\\"></audio></div>
  </div>
</div>"
`;
